### PR TITLE
disable building channelz if gRPC_USE_PROTO_LITE

### PR DESCRIPTION
**Pull request by yorkie**
See https://github.com/grpc/grpc/issues/19473

### CODE REVIEW

**Comment by hcaseyal** [comment]
@jtattermusch I'd like your input on https://github.com/grpc/grpc/issues/19472 before considering the changes suggested in this PR

**Comment by yorkie** [comment]
@hcaseyal Thanks for your reviews, and I have achieved the followings:

- signed the GRPC  CLA
- removed the previous changes on auto-generated contents
- introduced a new attribute `extra_cond` for libs, and updated the CMakeFile template
  - updated the `build.yaml` for building channelz only if not `gRPC_USE_PROTO_LITE`

**Comment by yorkie** [comment]
I still have a question, why do we maintain the generated file on the tree? Because I'm confused that if should generate once again those files and push them on this PR?

File: build.yaml [hunk 1]
```diff
@@ -2110,6 +2110,8 @@ libs:
   - grpc
   filegroups:
   - grpcpp_channelz_proto
+  extra_cond:
```
Comment (jtattermusch) [inline build.yaml]: This seems pretty adhoc, and also in the long run we'll be getting rid of build.yaml, so it doesn't make sense to add new properties like this.
We could just specialcase directly in CMakeLists.txt.template if needed, but I'm not sure it's worth doing it considering that gRPC_USE_PROTO_LITE is probably used only by few.

**Comment by jtattermusch** [comment]
It looks like due to https://github.com/grpc/grpc/issues/19472, `protoc` won't work when gRPC_USE_PROTO_LITE is enabled anyway (generated .proto code is needed for channelz_service to build), so this PR cannot really fix the problem?

**Comment by jtattermusch** [comment]
I posted a workaround in https://github.com/grpc/grpc/issues/19473#issuecomment-552382027 (and I think this PR brings too much complexity for something that not really used by many).

**Comment by jtattermusch** [comment]
Closing in favor of https://github.com/grpc/grpc/pull/21011

**Pull request closed without merging**